src/transaction/log_replication.cpp

```diff
@@ -280,15 +280,21 @@ namespace cublog
   {
     // skip calculation if bogus input (sometimes, it is -1);
     // TODO: fix bogus input at the source if at all possible (debugging revealed that
-    // it happens for LOG_COMMIT messages only)
+    // it happens for LOG_COMMIT messages only and there is no point at the source where the 'at_time'
+    // is not filled in; the only other possibility is a reading misalignment that remains below the radar)
     if (a_start_time_msec > 0)
       {
 	const int64_t end_time_msec = util_get_time_as_ms_since_epoch ();
 	const int64_t time_diff_msec = end_time_msec - a_start_time_msec;
-	assert (time_diff_msec > 0);
+	assert (time_diff_msec >= 0);
 
 	perfmon_set_stat (thread_p, PSTAT_REDO_REPL_DELAY, static_cast<int> (time_diff_msec), false);
 
+	if (prm_get_bool_value (PRM_ID_LOG_CALC_REPL_DELAY))
+	  {
+	    er_log_debug (ARG_FILE_LINE, "[CALC_REPL_DELAY]: %9lld msec", time_diff_msec);

```

Comment:
I think It is more helpful to you to use '_er_ log_debug()' than 'er log_debug()'. It is just a comment and you can choose a better one.

Reply:
er_log_debug requires PRM_ID_ER_LOG_DEBUG to be true. It is the generic debug logging function. Since this debug message already requires PRM_ID_LOG_CALC_REPL_DELAY to be true, \_er_log_debug may be called directly.